File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/session_retention_configuration.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class SessionRetentionConfiguration(object):
    """
    Session data retention configuration for agent flow
    """

    def __init__(self, **kwargs):
        """
        Initializes a new SessionRetentionConfiguration object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param retention_period_in_days:
            The value to assign to the retention_period_in_days property of this SessionRetentionConfiguration.
        :type retention_period_in_days: int

        :param session_size_limit:
            The value to assign to the session_size_limit property of this SessionRetentionConfiguration.
        :type session_size_limit: int

        :param thread_count_limit:
            The value to assign to the thread_count_limit property of this SessionRetentionConfiguration.
        :type thread_count_limit: int

        """
        self.swagger_types = {
            'retention_period_in_days': 'int',
            'session_size_limit': 'int',
            'thread_count_limit': 'int'
        }

        self.attribute_map = {
            'retention_period_in_days': 'retentionPeriodInDays',
            'session_size_limit': 'sessionSizeLimit',
            'thread_count_limit': 'threadCountLimit'
        }

        self._retention_period_in_days = None
        self._session_size_limit = None
        self._thread_count_limit = None

    @property
    def retention_period_in_days(self):
        """
        Gets the retention_period_in_days of this SessionRetentionConfiguration.
        No. of days session data will be kept


        :return: The retention_period_in_days of this SessionRetentionConfiguration.
        :rtype: int
        """
        return self._retention_period_in_days

    @retention_period_in_days.setter
    def retention_period_in_days(self, retention_period_in_days):
        """
        Sets the retention_period_in_days of this SessionRetentionConfiguration.
        No. of days session data will be kept


        :param retention_period_in_days: The retention_period_in_days of this SessionRetentionConfiguration.
        :type: int
        """
        self._retention_period_in_days = retention_period_in_days

    @property
    def session_size_limit(self):
        """
        Gets the session_size_limit of this SessionRetentionConfiguration.
        Max storage allocated to session data (in MB).


        :return: The session_size_limit of this SessionRetentionConfiguration.
        :rtype: int
        """
        return self._session_size_limit

    @session_size_limit.setter
    def session_size_limit(self, session_size_limit):
        """
        Sets the session_size_limit of this SessionRetentionConfiguration.
        Max storage allocated to session data (in MB).


        :param session_size_limit: The session_size_limit of this SessionRetentionConfiguration.
        :type: int
        """
        self._session_size_limit = session_size_limit

    @property
    def thread_count_limit(self):
        """
        Gets the thread_count_limit of this SessionRetentionConfiguration.
        Maximum no. of user prompt and agent response pairs per session


        :return: The thread_count_limit of this SessionRetentionConfiguration.
        :rtype: int
        """
        return self._thread_count_limit

    @thread_count_limit.setter
    def thread_count_limit(self, thread_count_limit):
        """
        Sets the thread_count_limit of this SessionRetentionConfiguration.
        Maximum no. of user prompt and agent response pairs per session


        :param thread_count_limit: The thread_count_limit of this SessionRetentionConfiguration.
        :type: int
        """
        self._thread_count_limit = thread_count_limit

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/SessionRetentionConfiguration.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * Session data retention configuration for agent flow
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=SessionRetentionConfiguration.Builder.class)

public final class SessionRetentionConfiguration  {
    @Deprecated
    @java.beans.ConstructorProperties({"retentionPeriodInDays", "sessionSizeLimit", "threadCountLimit"})
    public SessionRetentionConfiguration(Integer retentionPeriodInDays, Integer sessionSizeLimit, Integer threadCountLimit) {
        super();
        this.retentionPeriodInDays = retentionPeriodInDays;
        this.sessionSizeLimit = sessionSizeLimit;
        this.threadCountLimit = threadCountLimit;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * No. of days session data will be kept
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("retentionPeriodInDays")
        private Integer retentionPeriodInDays;

                /**
         * No. of days session data will be kept
         * @param retentionPeriodInDays the value to set
         * @return this builder
         **/
        

        public Builder retentionPeriodInDays(Integer retentionPeriodInDays) {
        this.retentionPeriodInDays = retentionPeriodInDays;
        return this;
        }
            /**
     * Max storage allocated to session data (in MB).
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("sessionSizeLimit")
        private Integer sessionSizeLimit;

                /**
         * Max storage allocated to session data (in MB).
         * @param sessionSizeLimit the value to set
         * @return this builder
         **/
        

        public Builder sessionSizeLimit(Integer sessionSizeLimit) {
        this.sessionSizeLimit = sessionSizeLimit;
        return this;
        }
            /**
     * Maximum no. of user prompt and agent response pairs per session
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("threadCountLimit")
        private Integer threadCountLimit;

                /**
         * Maximum no. of user prompt and agent response pairs per session
         * @param threadCountLimit the value to set
         * @return this builder
         **/
        

        public Builder threadCountLimit(Integer threadCountLimit) {
        this.threadCountLimit = threadCountLimit;
        return this;
        }


        public SessionRetentionConfiguration build() {
            SessionRetentionConfiguration model = new SessionRetentionConfiguration(this.retentionPeriodInDays
                    , this.sessionSizeLimit
                    , this.threadCountLimit);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(SessionRetentionConfiguration model) {
                this.retentionPeriodInDays(model.getRetentionPeriodInDays());
    this.sessionSizeLimit(model.getSessionSizeLimit());
    this.threadCountLimit(model.getThreadCountLimit());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * No. of days session data will be kept
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("retentionPeriodInDays")
    private final Integer retentionPeriodInDays;

        /**
     * No. of days session data will be kept
     * @return the value
     **/
    
    public Integer getRetentionPeriodInDays() {
        return retentionPeriodInDays;
    }


        /**
     * Max storage allocated to session data (in MB).
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("sessionSizeLimit")
    private final Integer sessionSizeLimit;

        /**
     * Max storage allocated to session data (in MB).
     * @return the value
     **/
    
    public Integer getSessionSizeLimit() {
        return sessionSizeLimit;
    }


        /**
     * Maximum no. of user prompt and agent response pairs per session
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("threadCountLimit")
    private final Integer threadCountLimit;

        /**
     * Maximum no. of user prompt and agent response pairs per session
     * @return the value
     **/
    
    public Integer getThreadCountLimit() {
        return threadCountLimit;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("SessionRetentionConfiguration(");
        sb.append("retentionPeriodInDays=").append(String.valueOf(this.retentionPeriodInDays));
        sb.append(", sessionSizeLimit=").append(String.valueOf(this.sessionSizeLimit));
        sb.append(", threadCountLimit=").append(String.valueOf(this.threadCountLimit));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof SessionRetentionConfiguration)) {
            return false;
        }

        SessionRetentionConfiguration other = (SessionRetentionConfiguration) o;
        return java.util.Objects.equals(this.retentionPeriodInDays, other.retentionPeriodInDays) &&
            java.util.Objects.equals(this.sessionSizeLimit, other.sessionSizeLimit) &&
            java.util.Objects.equals(this.threadCountLimit, other.threadCountLimit);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.retentionPeriodInDays == null ? 43 : this.retentionPeriodInDays.hashCode());
        result = (result * PRIME) + (this.sessionSizeLimit == null ? 43 : this.sessionSizeLimit.hashCode());
        result = (result * PRIME) + (this.threadCountLimit == null ? 43 : this.threadCountLimit.hashCode());
        return result;
    }


}
